app/src/main/java/com/mybus/activity/DisplayFavoritesActivity.java

```diff
@@ -1,11 +1,94 @@
 package com.mybus.activity;
 
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.TextView;
+
+import com.google.android.gms.maps.model.LatLng;
 import com.mybus.R;
+import com.mybus.adapter.FavoriteViewAdapter;
+import com.mybus.dao.FavoriteLocationDao;
+import com.mybus.listener.FavoriteAddListener;
+import com.mybus.listener.FavoriteListItemListener;
+import com.mybus.listener.FavoriteEditOldListener;
+import com.mybus.model.FavoriteLocation;
+import com.mybus.model.SearchType;
+import com.mybus.view.FavoriteNameAlertDialog;
+
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
 
 /**
  * Created by Julian Gonzalez <jgonzalez@devspark.com>
  */
-public class DisplayFavoritesActivity extends BaseDisplayActivity {
+public class DisplayFavoritesActivity extends BaseDisplayActivity implements FavoriteListItemListener, FavoriteAddListener, FavoriteEditOldListener {
```

Comment:
Why do you need two listeners for edit? one is onFavoriteItemEdit inside FavoriteListItemListener and the other is FavoriteEditOldLIstener